src/Page.js

```diff
@@ -581,6 +584,7 @@ Page.prototype.insertSiteNav = function (pageData) {
   if (siteNavContent === '') {
     return pageData;
   }
+  this.hasSiteNav = true;
```

Comment:
Since you only modify the flag to be true at this point. If you delete the `navigation.md` (Or blank the contents) from the layouts folder, the `site-nav` will still appear in the `markbind serve` preview. Even though MarkBind correctly detected `navigation.md` has changed.

The only correct behaviour in this code is changing `navigation.md` slightly and the changes will reflect in the `markbind serve` preview.

**Suggestion**
Explicitly set the flag to false in the if statements above for it to function properly when in preview mode.